pkg/asset/manifests/ingress.go

```diff
@@ -135,30 +135,28 @@ func (ing *Ingress) generateClusterConfig(config *types.InstallConfig) ([]byte,
 }
 
 func (ing *Ingress) generateDefaultIngressController(config *types.InstallConfig) ([]byte, error) {
-	switch config.Publish {
-	case types.InternalPublishingStrategy:
-		obj := &operatorv1.IngressController{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: operatorv1.GroupVersion.String(),
-				Kind:       "IngressController",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: "openshift-ingress-operator",
-				Name:      "default",
-			},
-			Spec: operatorv1.IngressControllerSpec{
-				EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
-					Type: operatorv1.LoadBalancerServiceStrategyType,
-					LoadBalancer: &operatorv1.LoadBalancerStrategy{
-						Scope: operatorv1.InternalLoadBalancer,
-					},
+	if config.Publish != types.InternalPublishingStrategy && config.InternalResources != nil && !config.InternalResources.Ingress {
```

Comment:
Could you please add a comment / debug statement explaining the conditions being checked here?

Reply:
Haven't gotten to testing the ingress private side. I just set it to do the same thing for `publish: Internal` but I'm assuming we might need to do something else here? If that even makes sense?

But the logic here should be if it's either `publish: Internal` or if the ingress is set to true then return nil. Looks like the check should just be `config.InternalResources.Ingress` but I don't think I should be taking this approach anyway.